Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.h

```diff
@@ -149,8 +148,9 @@ class SourceBufferPrivateRemote final
 
     std::atomic<uint64_t> m_totalTrackBufferSizeInBytes = { 0 };
 
-    bool isGPURunning() const { return !m_disconnected; }
-    bool m_disconnected { false };
+    bool isGPURunning() const { return !m_shutdown && !m_removed; }
+    std::atomic<bool> m_shutdown { false };
+    bool m_removed { false };
```

Comment:
m_isRemoved?
Also, I am not exactly sure why m_shutdown is atomic but not m_removed.
They seem to be accessed from both main thread (callOnMainThreadAndWait) and work queue

Reply:
m_removed is only accessed in the dispatcher; m_shutdown is immediately accessed in the receiver without a redispatch. but for consistency I'll make it atomic too